### DISCUSSION TITLE
Array partially initialize

### DISCUSSION THREAD

**Discussion by pengweiqhca** [dotnet/csharplang#7294] [General]
``` C#
var array = new int[condition ? 4 : 3] { 0, 1, 2 }; // CS0747

if (condition)
{
    array[3] = 3; 
}
```

**Comment by huoyaoyuan** [comment]
When trying this I got CS0150: constant value expected.
The compiler doesn't try to prove the conditional allocation is enough for the collection initializer.

With upcoming collection literal feature you may write:

```
int[] array = [ 0, 1, 2, ..(condition ? [3] : [])];
```
